Web/tues_pairs/src/components/Pages/Account/edit_personal_info.jsx

```diff
@@ -236,6 +239,14 @@ class EditPersonalInfo extends Component{
 
         const hasImage = photoURL ? true : false;
 
+        let valid = true;
+
+        providerData.forEach(provider => {
+            if(provider.providerId !== 'password') {
+                valid = false;
+            }
+        });
```

Comment:
Can you not extract this in ComponentDidMount() or one of the methods that trigger before the build state? I presume you only need this validation once for the state and not in each component render.

Also, I suppose you're checking whether the provider Id from the user's provider data matches the one for Firebase e-mail auth ("password") and setting the valid flag accordingly, correct?

Reply:
Yes you are right, I moved it in the componentDidMount() method. Thanks for the request!